### PR TITLE
Add runs endpoint

### PR DESCRIPTION
**Pull request by lamppu**
Please see individual commits for details.

Resolves https://github.com/eclipse-apoapsis/ort-server/issues/1074.

### CODE REVIEW

**Comment by Etsija** [comment]
If this resolves #1074, it should be mentioned in the description.

**Comment by sschuberth** [comment]
Please rebase to resolve a merge conflict.

**Comment by sschuberth** [comment]
> If this resolves #1074, it should be mentioned in the description.

Actually not in the PR description, but in the commit message.

**Comment by Etsija** [comment]
> > If this resolves #1074, it should be mentioned in the description.
> 
> Actually not in the PR description, but in the commit message.

Really? This PR consists of five commits, which together resolve the issue. In these cases, I've always put the "Resolves..." text to the PR description, because no individual commit of those five is responsible for resolving the issue. Can you elaborate why we would tag only one commit (and which? The latest one?) for resolving the issue?

**Comment by sschuberth** [comment]
> Really?

Yes, really. 

> Can you elaborate why we would tag only one commit, and which one should it be?

Assume that all but the last commit were already there, maybe created by someone else. Then logically it would still be the last commit that concludes the resolution of the issue.

Also, it does not make sense to sometimes resolve issues in commits and other times in PRs. To have one easy to remember rule, we align on the former. 

> I understand that PR descriptions are not seen in the commit history, so it would make sense to refer to issues being resolved in commit messages, but when looking at commit history, I see varying approaches to this, so maybe it should be discussed with the team.

This is not a team decision, but a project lead decision, IMO. So @mnonnenmacher has the last say. I tried to point it out in all PRs that I reviewed if the resolution was done in the PR instead of the commit. Could be that I missed some occurrences. But if it was not done in the commit, it's a mistake, IMO.

**Comment by mnonnenmacher** [comment]
> > Really?
> 
> Yes, really.
> 
> > Can you elaborate why we would tag only one commit, and which one should it be?
> 
> Assume that all but the last commit were already there, maybe created by someone else. Then logically it would still be the last commit that concludes the resolution of the issue.
> 
> Also, it does not make sense to sometimes resolve issues in commits and other times in PRs. To have one easy to remember rule, we align on the former.

In my experience it's sufficient to only link the commit that does the final change to fix an issue, like the last one in this PR, this provides enough context already and it's easy to find the PR from there. However, you could also link the previous commits by using "Relates to #12345" if you think that is important, I sometimes do this when I only partly fix an issue. To me adding the link to the PR description is optional but adding it to the commit is mandatory.

> > I understand that PR descriptions are not seen in the commit history, so it would make sense to refer to issues being resolved in commit messages, but when looking at commit history, I see varying approaches to this, so maybe it should be discussed with the team.
> 
> This is not a team decision, but a project lead decision, IMO. So @mnonnenmacher has the last say. I tried to point it out in all PRs that I reviewed if the resolution was done in the PR instead of the commit. Could be that I missed some occurrences. But if it was not done in the commit, it's a mistake, IMO.

The definition of the role is mainly about organizational responsibilities, but a project lead is not automatically a technical lead:
https://www.eclipse.org/projects/handbook/#roles-pl
It specifically says:

> A project lead is not necessary a technical lead. We have no formal designation of technical lead; this tends to occur organically.

This matches with my experience that the people who are most active in contributing and reviewing code automatically have more influence on technical decisions.

File: api/v1/model/src/commonMain/kotlin/FilterOperatorAndValue.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+data class FilterOperatorAndValue<T>(
+    val operator: SqlOperator,
+    val value: T
+)
+
+@Serializable
+/**
+ * An enumeration of SQL operators.
+ */
+enum class SqlOperator {
```
Comment (mnonnenmacher) [inline api/v1/model/src/commonMain/kotlin/FilterOperatorAndValue.kt:37]: I think I would prefer a more generic name like `ComparisonOperator`, at least in the API module. Mentioning SQL here exposes an implementation detail and they might in future also be used for requests that do get their data from the database, like the user endpoints.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

**Comment by sschuberth** [comment]
I believe you might need to rebase once more to make the new prettier checks work property.

**Review by mnonnenmacher — APPROVED** [review]
Very nice PR!
@lamppu  You need to rebase once more because there are conflicts.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**